Tests/AsyncHTTPClientTests/HTTPRequestStateMachineTests.swift

```diff
@@ -522,6 +525,27 @@ class HTTPRequestStateMachineTests: XCTestCase {
         XCTAssertEqual(state.channelInactive(), .wait)
     }
 
+    func testNIOSSLErrorUncleanShutdownShouldBeTreatedAsRemoteConnectionCloseWhileInWaitingForHeadState() {
+        var state = HTTPRequestStateMachine(isChannelWritable: true)
+        let requestHead = HTTPRequestHead(version: .http1_1, method: .GET, uri: "/")
+        let metadata = RequestFramingMetadata(connectionClose: false, body: .fixedSize(0))
+        XCTAssertEqual(state.startRequest(head: requestHead, metadata: metadata), .sendRequestHead(requestHead, startBody: false))
+
+        XCTAssertEqual(state.errorHappened(NIOSSLError.uncleanShutdown), .wait)
+        XCTAssertEqual(state.channelInactive(), .failRequest(HTTPClientError.remoteConnectionClosed, .none))
+    }
+
+    func testArbitraryErrorShouldBeTreatedAsARequestFailureWhileInWaitingForHeadState() {
+        struct ArbitraryError: Error {}
+        var state = HTTPRequestStateMachine(isChannelWritable: true)
+        let requestHead = HTTPRequestHead(version: .http1_1, method: .GET, uri: "/")
+        let metadata = RequestFramingMetadata(connectionClose: false, body: .fixedSize(0))
+        XCTAssertEqual(state.startRequest(head: requestHead, metadata: metadata), .sendRequestHead(requestHead, startBody: false))
+
+        XCTAssertEqual(state.errorHappened(ArbitraryError()), .failRequest(ArbitraryError(), .close))
```

Comment:
The error is not included in the Equatable conformance. To make sure you will need to unwrap the enum with a guard and make the error equatable.

Reply:
okay... but we have 18 places where we need to fix this.

Reply:
doing it correctly here is a start ;)